### PR TITLE
Questions for Dec 29 & Dec 30

### PR DESCRIPTION
**Pull request by Harihar98**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - [Introduction to matrices](https://en.wikipedia.org/wiki/Matrix_(mathematics))
 
 
+## **December 29 - Cup and String Phone**
+  - **Problem :**
+    - Tharun is playing with his friends using a string phone. To make this interesting he asked his friends to join him. He asked them to bring cup of their own and he will bring the strings for all of them so that all can communicate with each other. Now he wants to find the number of strings required to communicate with his friends. Help him with a program that takes a number of friends as input and give the number of strings as output.
+
+    For example Tharun plays with 2 of his friends. Then he will bring 3 Strings.This is because 1 string is enough to connect 2 friends.
```
Comment (akashsara) [inline README.md]: Explain it fully. Don't let the reader assume the answer. Clearly state why 3 are required to connect them together. As in say that each friend needs to be able to contact the other friends separately and all that.

File: README.md [hunk 2]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - [Introduction to matrices](https://en.wikipedia.org/wiki/Matrix_(mathematics))
 
 
+## **December 29 - Cup and String Phone**
+  - **Problem :**
+    - Tharun is playing with his friends using a string phone. To make this interesting he asked his friends to join him. He asked them to bring cup of their own and he will bring the strings for all of them so that all can communicate with each other. Now he wants to find the number of strings required to communicate with his friends. Help him with a program that takes a number of friends as input and give the number of strings as output.
```
Comment (akashsara) [inline README.md]: If he is playing with his friends, why did he ask his friends to join him? Seems redundant. Remove the asked his friends to join him sentence.

File: README.md [hunk 3]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    For example Tharun plays with 2 of his friends. Then he will bring 3 Strings.This is because 1 string is enough to connect 2 friends.
+
+    <img src="/src/docs/dec29-string-phone.jpg" width="450" height="300">
+
+  - **Examples :**
+  
+    1) Input : 7\
+       Output : 28\
+       Explanation : \
+            Tharun & his 6 friends come together to play this game. So to connect all of 7 cups, we need 28 strings.\
```
Comment (akashsara) [inline README.md]: This isn't really an explanation. This is just stating the input and output in word form.

File: README.md [hunk 4]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+
+    2) Input : 10\
+       Output : 55
+ 
+  - **Uses :**
```
Comment (akashsara) [inline README.md]: Explain how it's used here.

Comment (akashsara) [reply README.md]: As in what a triangular series is/where it's used. How computers network using this and so on.

File: README.md [hunk 5]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+
+  - **Resources :**
+    - [Triangular Series](https://en.wikipedia.org/wiki/Triangular_number)
+
+
+## **December 30 - Number of Diagonals**
+  - **Problem :**
+    - There is a n-sided polygon. Find the number of [diagonals](https://en.wikipedia.org/wiki/Diagonal) that are possible. Do not count duplicates lines. Diagonals are Line segments that connect the non-adjacent vertices.
+
+  - **Example :**
```
Comment (akashsara) [inline README.md]: Put the smallest examples at the start. Explain how you get that output. We want to make it clear and obvious. So put the rectangle at the start. Say that a rectangle has two diagonals, or better, show a diagram. Same for the pentagon.

File: README.md [hunk 6]
```diff
@@ -763,6 +765,59 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    2) Input : 11\
+       Output : 44
+
+    3) Input : 4 (Quadrilateral - Rectangle)\
+       Output : 2
+
+    4) Input : 5 (Pentagon)\
+       Output : 5
+
+  - **Uses :**
```
Comment (akashsara) [inline README.md]: Explain how it's used in each case.

File: README.md (hunk 2 above)
Comment (Harihar98) [reply README.md]: done. fixed.

File: README.md (hunk 1 above)
Comment (Harihar98) [reply README.md]: done

Comment (Harihar98) [reply README.md]: done

File: README.md [hunk 7]
```diff
@@ -763,6 +765,62 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+  
+    1) Input : 5\
+       Output : 10\
+       <img src="/src/docs/dec29-mesh.jpg">\
+       Here 5 boxes represent Tharun & his 4 friends and the connections are the strings. Each friend needs to be able to contact the other friends separately, so we need 10 strings to connect them all.
+
+    1) Input : 7\
+       Output : 21\
+       Explanation : \
+            Tharun & his 6 friends come together to play this game. So to connect all of 7 cups, we need 28 strings.\
```
[the remark this answers is not in the document]
Comment (Harihar98) [reply README.md]: rectified.

File: README.md (hunk 1 above)
Comment (Harihar98) [reply README.md]: modified

**Pull request merged**